File: C++/Medium/WrongAnswer/number-of-subarrays-with-bounded-maximum_4.cpp
class Solution {
public:
    int numSubarrayBoundedMax(vector<int>& nums, int left, int right) {
        
        int res = 0;
        int l = 0, r = 0;
        int n = nums.size();
        while (r < n) {
            if (nums[r] < left) {
                res += r - l;
            } else if (nums[r] > right) {
                l = r + 1;
            } else {
                int tmp = r;
                while (tmp < n && nums[tmp] <= right) {
                    tmp++;
                }
                res += (tmp - r) * (r - l + 1);
                l = r = tmp;
            }
            r++;
        }
        return res;
    }
};

File: Java/Medium/WrongAnswer/number-of-subarrays-with-bounded-maximum_4.java
class Solution {
    public int numSubarrayBoundedMax(int[] nums, int left, int right) {
        
        int count = 0;
        int n = nums.length;
        int[] dp = new int[n];
        for (int i = 0; i < n; i++) {
            if (nums[i] >= left && nums[i] <= right) {
                dp[i] = i > 0 ? dp[i - 1] + 1 : 1;
                count += dp[i];
            } else if (nums[i] < left) {
                dp[i] = i > 0 ? dp[i - 1] : 0;
                count += dp[i];
            }
        }
        return count;
    }
}